runtime/src/main/kotlin/kotlin/Arrays.kt

```diff
@@ -146,12 +148,12 @@ public final class ShortArray {
      */
     @InlineConstructor
     public constructor(size: Int, init: (Int) -> Short): this(size) {
-        for (i in 0..size - 1) {
+        for (i in 0 until this.size) {
             this[i] = init(i)
         }
     }
 
-    public val size: Int
+    inline public val size: Int
```

Comment:
Better be handled by codegen, I guess, and actually may make optimizer understand what's going on easier.

Reply:
Before doing this PR I tried to some intrinsics but I had some strange errors about a symbol not found, so maybe I did it wrong.